semgrep-core/src/parsing/pfff/java_to_generic.ml

```diff
@@ -421,6 +421,15 @@ and arguments v : G.argument list G.bracket = bracket (list argument) v
 
 and fix_op v = H.conv_incr v
 
+and resource t (v : resource) : G.stmt =
+  match v with
+  | Left v ->
+      let ent, v = var_with_init v in
+      G.DefStmt (ent, G.VarDef v) |> G.s
+  | Right e -> G.ExprStmt (expr e, t) |> G.s
+
+and resources (t1, v, t2) = G.Block (t1, list (resource t2) v, t2) |> G.s
```

Comment:
What happens to this code:
```
    try (
        java.util.zip.ZipFile zf =
             new java.util.zip.ZipFile(zipFileName);
        java.io.BufferedWriter writer = 
            java.nio.file.Files.newBufferedWriter(outputFilePath, charset)
    ) {
        // Enumerate each entry
        for (java.util.Enumeration entries =
                                zf.entries(); entries.hasMoreElements();) {
            // Get the entry name and write it to the output file
            String newLine = System.getProperty("line.separator");
            String zipEntryName =
                 ((java.util.zip.ZipEntry)entries.nextElement()).getName() +
                 newLine;
            writer.write(zipEntryName, 0, zipEntryName.length());
        }
    }
```
(found on the Java documentation at https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html)

Reply:
E.g., can we have a test that matches against multiple resources?

Reply:
It's a block of resources (the semicolon token is lost, may be a problem one day). I added a test to cover it